python/lsst/ts/mtdome/mock_llc/mock_motion/base_llc_motion_with_crawl.py

```diff
@@ -0,0 +1,173 @@
+# This file is part of ts_mtdome.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import math
+from abc import ABC
+
+from ...enums import LlcMotionState
+from .base_llc_motion import BaseLlcMotion
+
+
+class BaseLlcMotionWithCrawl(BaseLlcMotion, ABC):
+    def __init__(
+        self,
+        start_position: float,
+        min_position: float,
+        max_position: float,
+        max_speed: float,
+        start_tai: float,
+    ):
+        super().__init__(
+            start_position=start_position,
+            min_position=min_position,
+            max_position=max_position,
+            max_speed=max_speed,
+            start_tai=start_tai,
+        )
+
+        # When a move or crawl command is received, it specifies the crawl
+        # crawl_velocity.
+        self._crawl_velocity = 0.0
+
+    def _get_duration(self) -> float:
+        """Determines the duration of the move using the distance of the move
+        and the maximum speed, or zero in case of a crawl.
+
+        Returns
+        -------
+        duration: `float`
+            The duration of the move, or zero in case of a crawl.
+        """
+        duration = super()._get_duration()
+        if self._commanded_motion_state == LlcMotionState.CRAWLING:
+            # A crawl command is executed instantaneously.
+            duration = 0
+        return duration
```

Comment:
Why call super()._get_duration() before testing if crawling? Does it change internal state or do some error checking? Please consider adding a code comment explaining, since the naive thing to do is only call super()._get_duration() if not crawling, and readers will want to know.

Reply:
The super method computes the duration for the default case, which is MOVING. This class extends that for CRAWLING. I have added comments to explain that.